framework/infrastructure/traffic_director.py

```diff
@@ -679,32 +724,60 @@ def find_unused_forwarding_rule_port(
         # TODO(sergiitk): custom exception
         raise RuntimeError("Couldn't find unused forwarding rule port")
 
-    def create_forwarding_rule(self, src_port: int):
-        name = self.make_resource_name(self.FORWARDING_RULE_NAME)
+    def create_forwarding_rule(
+        self, src_port: int, ipv6=False, ip_address="0.0.0.0"
+    ):
+        if ipv6:
+            name = self.make_resource_name(self.FORWARDING_RULE_NAME_IPV6)
+            target_proxy = self.target_proxy_v6
+        else:
+            name = self.make_resource_name(self.FORWARDING_RULE_NAME)
+            target_proxy = self.target_proxy
+
         src_port = int(src_port)
         logging.info(
-            'Creating forwarding rule "%s" in network "%s": 0.0.0.0:%s -> %s',
+            'Creating forwarding rule "%s" in network "%s": %s:%s -> %s',
             name,
             self.network,
             src_port,
-            self.target_proxy.url,
+            ip_address,
+            target_proxy.url,
         )
         resource = self.compute.create_forwarding_rule(
-            name, src_port, self.target_proxy, self.network_url
+            name,
+            src_port,
+            target_proxy,
+            self.network_url,
+            ip_address=ip_address,
         )
-        self.forwarding_rule = resource
+
+        if ipv6:
+            self.forwarding_rule_v6 = resource
+        else:
+            self.forwarding_rule = resource
+
         return resource
 
-    def delete_forwarding_rule(self, force=False):
+    def delete_forwarding_rule(self, force=False, ipv6=False):
         if force:
-            name = self.make_resource_name(self.FORWARDING_RULE_NAME)
+            if ipv6:
+                name = self.make_resource_name(self.FORWARDING_RULE_NAME_IPV6)
+            else:
+                name = self.make_resource_name(self.FORWARDING_RULE_NAME)
+        elif ipv6:
+            if self.forwarding_rule_v6:
+                name = self.forwarding_rule_v6.name
```

Comment:
Ditto, should we consider what should be the `name` here in case `self.forwarding_rule_v6` is not true?